src/validation.cpp

```diff
@@ -945,6 +945,19 @@ bool AcceptToMemoryPoolWorker(CTxMemPool& pool, CValidationState& state, const C
         if (!Params().RequireStandard()) {
             scriptVerifyFlags = GetArg("-promiscuousmempoolflags", scriptVerifyFlags);
         }
+        
+        // There are two approaches with regards to mempool behaviour and
+        // the new SIGHASH type. 
+        // (1) Reject until the fork. This can allow certain DDoS attacks,
+        //     where a node that is still catching up is sent a new transaction,
+        //     which can cause that node to get banned
+        // (2) Allow new transactions from day one, but don't include them in blocks
+        //     until actual fork. Eventually these transactions would drop out of
+        //     the mempool. 
+        const int nBlockHeight = chainActive.Height();
+        if (nBlockHeight >= Params().GetConsensus().SW2XHeight) {
+            scriptVerifyFlags |= SCRIPT_VERIFY_ALLOW_2X_SIGHASH;
+        }
```

Comment:
What about relaying anti-replay transactions to other nodes once they are in the mempool? If the `mempool` RPC returns these transactions to a non-upgraded btc1 node, wouldn't that node get upset because you're sending it invalid transactions? 

Perhaps the [mempool` RPC call](https://bitcoin.org/en/developer-reference#mempool) could exclude these transactions until a hardcoded date just before the fork? That gives node operators some time to upgrade.